object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/MustCallInvokedChecker.java

```diff
@@ -199,6 +202,94 @@ private void handleInvocation(Set<ImmutableSet<LocalVarWithTree>> newDefs, Node
     }
   }
 
+  /**
+   * Checks that an invocation of a ResetMustCall method is valid. Such an invocation is valid if
+   * one of the following conditions is true: 1) the target is an owning pointer 2) the target is
+   * tracked in newdefs 3) the method in which the invocation occurs also has an @ResetMustCall
+   * annotation, with the same target
+   *
+   * <p>If none of the above are true, this method issues a reset.not.owning error.
+   *
+   * @param newDefs the local variables that have been defined in the current compilation unit (and
+   *     are therefore going to be checked later)
+   * @param node a method invocation node, invoking a method with a ResetMustCall annotation
+   */
+  private void checkResetMustCallInvocation(
+      Set<ImmutableSet<LocalVarWithTree>> newDefs, MethodInvocationNode node) {
+    AnnotationMirror resetMustCall =
+        typeFactory.getDeclAnnotation(
+            TreeUtils.elementFromUse(node.getTree()), ResetMustCall.class);
+    String targetStrWithoutAdaptation =
+        AnnotationUtils.getElementValue(resetMustCall, "value", String.class, true);
+    TreePath currentPath = typeFactory.getPath(node.getTree());
+    JavaExpressionContext context =
+        JavaExpressionParseUtil.JavaExpressionContext.buildContextForMethodUse(node, checker);
+    String targetStr =
+        MustCallTransfer.standardizeAndViewpointAdapt(
+            targetStrWithoutAdaptation, currentPath, context);
+    JavaExpression target;
+    try {
+      target = typeFactory.parseJavaExpressionString(targetStr, currentPath);
+    } catch (JavaExpressionParseException e) {
+      typeFactory
+          .getChecker()
+          .reportError(
+              node.getTree(),
+              "mustcall.not.parseable",
+              node.getTarget().getMethod().getSimpleName(),
+              targetStr);
+      return;
+    }
+
+    if (target instanceof LocalVariable) {
+
+      Element elt = ((LocalVariable) target).getElement();
+      if (typeFactory.getDeclAnnotation(elt, Owning.class) != null) {
+        // if the target is an Owning param, this satisfies case 1
+        return;
+      }
+
+      for (ImmutableSet<LocalVarWithTree> defAliasSet : newDefs) {
+        for (LocalVarWithTree localVarWithTree : defAliasSet) {
+          if (target.equals(localVarWithTree.localVar)) {
+            // satisfies case 2 above
+            return;
+          }
+        }
+      }
+    }
+    if (target instanceof FieldAccess) {
+      Element elt = ((FieldAccess) target).getField();
+      if (typeFactory.getDeclAnnotation(elt, Owning.class) != null) {
+        // if the target is an Owning field, this satisfies case 1
+        return;
+      }
+    }
+
+    MethodTree containingMethod = TreePathUtil.enclosingMethod(currentPath);
+    if (containingMethod != null) {
+      ExecutableElement enclosingMethod = TreeUtils.elementFromDeclaration(containingMethod);
```

Comment:
Variable names `containingMethod` and `enclosingMethod` are a bit confusing.  Maybe `enclosingMethod` and `enclosingMethodElement`?

Reply:
Thanks for the suggestion. I also made a similar change in another part of this PR that does something similar and used the same confusing naming scheme.